src/tests/Microsoft.Diagnostics.NETCore.Client/ReversedServerHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Runtime.InteropServices;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public static class ReversedServerHelper
+    {
+        /// <summary>
+        /// Creates a unique server name to avoid collisions from simultaneous running tests
+        /// or potentially abandoned socket files.
+        /// </summary>
+        public static string CreateServerTransportName()
+        {
+            string transportName = "DOTNET_DIAGSERVER_TESTS_" + Path.GetRandomFileName();
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return transportName;
+            }
+            else
+            {
+                return Path.Combine(Path.GetTempPath(), transportName);
+            }
+        }
+
+        /// <summary>
+        /// Starts the Tracee executable while enabling connection to reverse diagnostics server.
+        /// </summary>
+        public static TestRunner StartTracee(ITestOutputHelper _outputHelper, string transportName)
+        {
+            var runner = new TestRunner(CommonHelper.GetTraceePath(targetFramework: "net5.0"), _outputHelper);
+            AddReversedServer(runner, transportName);
+            runner.Start();
+            return runner;
+        }
+
+        public static void AddReversedServer(TestRunner runner, string transportName)
+        {
+            runner.AddEnvVar("DOTNET_DiagnosticsMonitorAddress", transportName);
```

Comment:
I don't see anywhere that you send the `ResumeRuntime` diagnostics IPC command.  Setting `DOTNET_DiagnosticsMonitorAddress` will cause the runtime to pause during startup and wait for a `ResumeRuntime` command on the IPC channel.  If you don't want the process to pause, you can set `DOTNET_DiagnosticsMonitorPauseOnStart=0` and the runtime won't pause.  Sending the command is a no-op if the target process is not paused, so you can send it pretty painlessly.

Reply:
The tests at this time do not pause, probably because they are using an earlier version of the runtime than the version that contains the change to wait. So maybe what I'll do now, at the very least, is set the `DOTNET_DiagnosticsMonitorPauseOnStart` env var so that they don't break. I don't know if I can send the `ResumeRuntime` message yet if they are still using an earlier version of the runtime without breaking the diagnostics communication.

Reply:
You can send the command to older runtimes, but you'll get back an error.  It shouldn't break anything, though.

Reply:
I'll do this in my next iteration.